chainercv/links/model/faster_rcnn/faster_rcnn.py

```diff
@@ -0,0 +1,321 @@
+                and the range of their values are :math:`[0, 255]`.
+
+        Returns:
+           tuple of lists:
+           This method returns a tuple of three lists,
+           :obj:`(bboxes, labels, scores)`.
+
+           * **bboxes**: A list of float arrays of shape :math:`(R, 4)`, \
+               where :math:`R` is the number of bounding boxes in a image. \
+               Each bouding box is organized by \
+               :obj:`(x_min, y_min, x_max, y_max)` \
+               in the second axis.
+           * **labels** : A list of integer arrays of shape :math:`(R,)`. \
+               Each value indicates the class of the bounding box. \
+               Values are in range :math:`[0, L - 1]`, where :math:`L` is the \
+               number of the foreground classes.
+           * **scores** : A list of float arrays of shape :math:`(R,)`. \
+               Each value indicates how confident the prediction is.
+
+        """
+        prepared_imgs = list()
+        scales = list()
+        for img in imgs:
+            _, H, W = img.shape
+            img = self.prepare(img.astype(np.float32))
+            scale = img.shape[2] / W
+            prepared_imgs.append(img)
+            scales.append(scale)
+
+        bboxes = list()
+        labels = list()
+        scores = list()
+        for img, scale in zip(prepared_imgs, scales):
+            img_var = chainer.Variable(
+                self.xp.asarray(img[None]), volatile=chainer.flag.ON)
+            H, W = img_var.shape[2:]
+            roi_cls_locs, roi_scores, rois, _ = self.__call__(
+                img_var, scale=scale, test=True)
+            # We are assuming that batch size is 1.
+            roi_cls_loc = roi_cls_locs.data
+            roi_score = roi_scores.data
+            roi = rois / scale
+
+            # Convert predictions to bounding boxes in image coordinates.
+            # Bounding boxes are scaled to the scale of the input images.
+            mean = self.xp.tile(self.xp.asarray(self.loc_normalize_mean),
+                                self._n_class)
+            std = self.xp.tile(self.xp.asarray(self.loc_normalize_std),
+                               self._n_class)
+            roi_cls_loc = (roi_cls_loc * std + mean).astype(np.float32)
+            roi_cls_loc = roi_cls_loc.reshape(-1, self._n_class, 4)
+            roi = self.xp.broadcast_to(roi[:, None], roi_cls_loc.shape)
+            cls_bbox = loc2bbox(roi.reshape(-1, 4), roi_cls_loc.reshape(-1, 4))
+            cls_bbox = cls_bbox.reshape(-1, self._n_class * 4)
+            # clip bounding box
+            cls_bbox[:, slice(0, 4, 2)] = self.xp.clip(
+                cls_bbox[:, slice(0, 4, 2)], 0, W / scale)
+            cls_bbox[:, slice(1, 4, 2)] = self.xp.clip(
+                cls_bbox[:, slice(1, 4, 2)], 0, H / scale)
+
+            prob = F.softmax(roi_score).data
+
+            raw_cls_bbox = cuda.to_cpu(cls_bbox)
+            raw_prob = cuda.to_cpu(prob)
```

Comment:
I found that GPU NMS is not as relatively fast in this part of the code because the number of boxes is small.

I think forcing CPU is fine. For better performance, we should cythonize CPU NMS.